File: simple-performance/src/main/java/com/example/simple_performance/user/service/UserService.java
package com.example.simple_performance.user.service;

import com.example.simple_performance.user.db.UserEntity;
import com.example.simple_performance.user.db.UserRepository;
import com.example.simple_performance.user.model.UserDto;
import com.example.simple_performance.user.model.UserRequest;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.time.LocalDateTime;
import java.util.List;
import java.util.stream.Collectors;


@Service
@RequiredArgsConstructor
public class UserService {
    private final UserRepository userRepository;
    private final UserConvertor userConvertor;

    public UserEntity create(UserRequest userRequest) {
        var userEntity = UserEntity.builder()
                .userName(userRequest.getUserName())
                .email(userRequest.getEmail())
                .password(userRequest.getPassword())
                .registeredAt(LocalDateTime.now())
                .build();

        return userRepository.save(userEntity);
    }

    public List<UserDto> findAll() {
        return userRepository
                .findAll()
                .stream()
                .map(userConvertor::toDto)
                .toList();
    }
}

File: kotlin-example/src/main/kotlin/ex12_mix/UserService.kt
package org.example.ex12_mix

import org.example.user.UserDtoMg
import java.time.LocalDateTime

fun main() {
    val userService = UserService()
    userService.logic()
}

class UserService {
    fun logic () {
        // java 가져오면 named parameter 안됨 ㅜㅜ
        val userDto = UserDtoMg(
           null,
            "foo@bar.com",
            11,
            "REGISTERED",
            LocalDateTime.now(),
            LocalDateTime.now(),
            LocalDateTime.now()
        )
        userDto.name?.let {println(it.length)} ?: println("null".length)
    }
}